### DISCUSSION TITLE
Options for vertical and horizontal terminal anchoring

### DISCUSSION THREAD

**Discussion by kevclark** [LazyVim/LazyVim#253] [Q&A] [answered]
Before transitioning over to LazyVim I used to use toggleterm.nvim in my workflow.  As LazyVim creates its own tooling around nvim's builtin terminal it didn't seem worth adding toggleterm.nvim back into the mix.  However features I really miss is being able to bring up the terminal and anchor it in either horizontal or vertical positions, so I can keep it there to run builds and test output, whist being able to hop back to the source window.

Is there anyway in LazyVim to add this easily to the util section?

Thanks

**Comment by dpetka2001** [comment]
How about `vsp +term`? Or the same for `horizontal` whatever you prefer.

**Reply by kevclark** [reply]
Yeah that works, but it will have the default 50/50 split and it doesn't automatically put you into insert mode. Small things I know, just things that were out of the box with toggleterm.nvim. It would be a nice addition to Lazy. But if not easy, I will just go back to toggle. No biggy.

**Reply by dpetka2001** [reply]
You can still resize the buffers with available `LazyVim` keymaps and you can create an `autocmd` to automatically enter `insert` mode if you like as discussed [here](https://vi.stackexchange.com/questions/3670/how-to-enter-insert-mode-when-entering-neovim-terminal-pane). You might want to use `BufEnter` or `TermOpen` to play around what works for you. 
Alternatively, you can just use `ToggleTerm` as well if that suits you more. I also use it for some of its features.

**Reply by kevclark** [reply]
Thanks, I'll explore.

**Comment by folke** [comment, marked as answer]
I only added the current terminal functionality, since it basically came for free with lazy.nvim.
For more advanced terminal usage, it's probably better to use toggleterm.
